src/nspawn/nspawn.c

```diff
@@ -3783,6 +3759,30 @@ static int outer_child(
         if (r < 0)
                 return r;
 
+        if (arg_userns_mode != USER_NAMESPACE_NO &&
```

Comment:
why move this down?

Reply:
To do this:
* https://github.com/systemd/systemd/pull/19828#issuecomment-855999595 
* https://github.com/systemd/systemd/pull/19828#issuecomment-858117042

But the change is not sufficient to fix the `EOVERFLOW`-error.
A workaround is to have all needed mount paths in the root filesystem already and not create any folders in the mount process.